torchtune/modules/peft/mora.py

```diff
@@ -0,0 +1,194 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+import math
+from typing import List
+
+import torch
+
+import torch.nn.functional as F
+
+from torch import nn, Tensor
+
+from torchao.dtypes.nf4tensor import linear_nf4, to_nf4
+from torchtune.modules.low_precision import _register_nf4_dispatch_ops  # noqa: F401
+from torchtune.modules.peft.peft_utils import AdapterModule
+
+
+class MoRALinear(nn.Module, AdapterModule):
+    """MoRA linear layer as introduced in `MoRA: High-Rank Updating for
+    Parameter-Efficient Fine-Tuning <https://arxiv.org/abs/2405.12130>`_.
```

Comment:
I remember we had talked about the 6 or so different implementations in kongds. Looks like here you're using the RoPE-inspired version. I thought a lot of the results in the paper use the group-based decomposition from equation 6, do you know if this version performs better? Mainly I want to make sure we are balancing providing the canonical version from the paper and (as much as possible) giving an implementation that's extensible to other cases.